pkg/comp-functions/functions/vshnforgejo/deploy.go

```diff
@@ -0,0 +1,204 @@
+
+	securityContext := map[string]any{}
+	if svc.Config.Data["isOpenshift"] == "true" {
+		securityContext = map[string]any{
+			"runAsUser":                nil,
+			"allowPrivilegeEscalation": false,
+			"podSecurityContext": map[string]any{
+				"fsGroup": nil,
+			},
+			"containerSecurityContext": map[string]any{
+				"runAsUser":                nil,
+				"allowPrivilegeEscalation": false,
+			},
+			"capabilities": map[string]any{
+				"drop": []string{
+					"ALL",
+				},
+			},
+		}
+	}
+
+	values := map[string]any{
+		"gitea": map[string]any{
+			"admin": map[string]any{
+				"existingSecret": secretName,
+			},
+			"config": map[string]any{
+				"admin": map[string]any{
+					"SEND_NOTIFICATION_EMAIL_ON_NEW_USER": true,
+				},
+				"cache": map[string]any{
+					"ADAPTER": "twoqueue",
+					"HOST":    "'{\"size\":100, \"recent_ratio\":0.25, \"ghost_ratio\":0.5}'",
+				},
+				"cron": map[string]any{
+					"ENABLED": true,
+					"archive_cleanup": map[string]any{
+						"SCHEDULE":   "@hourly",
+						"OLDER_THAN": "2h",
+					},
+				},
+				"database": map[string]any{
+					"DB_TYPE": "sqlite3",
+				},
+				"image": map[string]any{
+					"tag": comp.Spec.Parameters.Service.Version,
+				},
+				"indexer": map[string]any{
+					"ISSUE_INDEXER_TYPE":   "bleve",
+					"REPO_INDEXER_ENABLED": true,
+				},
+				"lfs": map[string]any{
+					"PATH": "/data/git/lfs",
+				},
+				"log": map[string]any{
+					"LEVEL": "info",
+				},
+				"packages": map[string]any{
+					"LIMIT_SIZE_CONTAINER": "2 GiB",
+				},
+				"queue": map[string]any{
+					"TYPE": "level",
+				},
+				"repository": map[string]any{
+					"ROOT": "/data/git/repositories",
+				},
+				"security": map[string]any{
+					"REVERSE_PROXY_TRUSTED_PROXIES": "*",
+				},
+				"server": map[string]any{
+					"DISABLE_SSH":      true,
+					"LANDING_PAGE":     "login",
+					"LFS_START_SERVER": true,
+					"MINIMUM_KEY_SIZE": true,
+					"OFFLINE_MODE":     true,
+				},
+				"session": map[string]any{
+					"PROVIDER": "memory",
+				},
+			},
+			"metrics": map[string]any{
+				"enabled": true,
+			},
+			"serviceMonitor": map[string]any{
+				"enabled": true,
+			},
+		},
+		"ingress": map[string]any{
+			"annotations": map[string]string{
+				"cert-manager.io/cluster-issuer": "letsencrypt-staging",
+			},
+			"enabled": true,
+			"hosts":   []map[string]any{},
+			"tls": []map[string]any{
+				{
+					"hosts":      []string{},
+					"secretName": "forgejo-tls",
+				},
+			},
+		},
+		"persistance": map[string]any{
+			"enabled": true,
+		},
+		"postgresql": map[string]any{
+			"enabled": false,
+		},
+		"postgresql-ha": map[string]any{
+			"enabled": false,
+		},
+		"redis": map[string]any{
+			"enabled": false,
+		},
+		"redis-cluster": map[string]any{
+			"enabled": false,
+		},
+		"strategy": map[string]any{
+			"type": "Recreate",
+		},
+	}
+
+	for _, host := range comp.Spec.Parameters.Service.FQDN {
```

Comment:
Maybe the `unstructured` package could help make this more readable?

Reply:
have You got a snippet?

Reply:
We use it quite a lot already: https://github.com/search?q=repo%3Avshn%2Fappcat%20unstructured&type=code

It's basically helper functions to work with such `map[string]any` objects.

here's the doc: https://pkg.go.dev/k8s.io/apimachinery@v0.30.2/pkg/apis/meta/v1/unstructured